src/pybind/mgr/dashboard/frontend/src/app/shared/api/user.service.ts

```diff
@@ -40,6 +45,14 @@ export class UserService {
     });
   }
 
+  validateUserName(user_name: string): Observable<boolean> {
+    return this.get(user_name).pipe(
+      mergeMap((resp: any) => {
+        return observableOf(resp.username === user_name);
+      })
+    );
```

Comment:
the calls inside pipe will only be called if the request is successful, so we can simply return true each time:
```suggestion
    return this.get(user_name, true).pipe(mapTo(true));
```

Reply:
It should also return a false value so that it can be used in the validator. Else the Validation part won't be working as intended.

Reply:
I think it ends up working, because the service method will receive the null value.

Reply:
Yes. Its working. But the only thing is Validation getting failed because `validateUserName` is not returning any Observable<boolean> object. 

Instead of returning null in the interceptor I returned the error and caught the error in the user-services file. Then using the catchError method I returned the Observable false object and then I can see the validation is getting successful. 
(Validation in the sense the green tick which appears on the Username box)